nav2_smac_planner/src/node_hybrid.cpp

```diff
@@ -434,12 +434,18 @@ float NodeHybrid::getTraversalCost(const NodePtr & child)
 
 float NodeHybrid::getHeuristicCost(
   const Coordinates & node_coords,
-  const Coordinates & goal_coords)
+  const CoordinateVector & goals_coords)
 {
+  // obstacle heuristic does not depend on goal heading
   const float obstacle_heuristic =
-    getObstacleHeuristic(node_coords, goal_coords, motion_table.cost_penalty);
-  const float dist_heuristic = getDistanceHeuristic(node_coords, goal_coords, obstacle_heuristic);
-  return std::max(obstacle_heuristic, dist_heuristic);
+    getObstacleHeuristic(node_coords, goals_coords[0], motion_table.cost_penalty);
+  float distance_heuristic = std::numeric_limits<float>::max();
+  for (unsigned int i = 0; i < goals_coords.size(); i++) {
```

Comment:
Ditto. It might be best to just use the main goal. If so, then this could be reverted to taking in the `Coordinates`.

Reply:
With the suggestion above, I can make this changes

Reply:
@SteveMacenski  With the below approach, this way I can just pass the coordinates instead of the coordinatevector, the only problem I see with this is https://github.com/stevedanomodolor/navigation2/blob/1a34ea3f3789aa66955a389b8db17acef95a836b/nav2_smac_planner/src/node_lattice.cpp#L422. I am sure how much of an influence it would be. 
```
 dist_heuristic_lookup_table.resize(size_lookup_total);
  dist_heuristic_lookup_table_bidirectional.resize(size_lookup_total);
  dist_heuristic_lookup_table_all_direction.resize(size_lookup_total);
```

Reply:
The `obstacle_heuristic == 0.0` case is defensive programming in case everything goes very wrong - this should not happen that both the cost is `0` and its outside of the window around the goal. This is a "just in case" case

Reply:
Then I will go with the approach I mentioned above and update the benchmarks again, this could reduce more time.